packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -511,6 +604,59 @@ export class EngineReportingAgent<TContext = any> {
     }
   }
 
+  public startSchemaReporting({
+    executableSchemaId,
+    executableSchema,
+  }: {
+    executableSchemaId: string;
+    executableSchema: string;
+  }) {
+    if (this.currentSchemaReporter) {
+      this.currentSchemaReporter.stop();
+    }
+
+    const serverInfo = {
+      bootId: this.bootId,
+      graphVariant: this.graphVariant,
+      // The infra environment in which this edge server is running, e.g. localhost, Kubernetes
+      // Length must be <= 256 characters.
+      platform: process.env.APOLLO_SERVER_PLATFORM || 'local',
+      runtimeVersion: `node ${process.version}`,
+      executableSchemaId: executableSchemaId,
+      // An identifier used to distinguish the version of the server code such as git or docker sha.
+      // Length must be <= 256 charecters
+      userVersion: process.env.APOLLO_SERVER_USER_VERSION,
+      // "An identifier for the server instance. Length must be <= 256 characters.
+      serverId:
+        process.env.APOLLO_SERVER_ID || process.env.HOSTNAME || os.hostname(),
+      libraryVersion: `apollo-engine-reporting@${
+        require('../package.json').version
+      }`,
+    };
+
+    // Jitter the startup between 0 and 10 seconds
+    const delay = Math.floor(
+      Math.random() *
+        (this.options.experimental_schemaReportingInitialDelayMaxMs || 10_000),
+    );
+
+    const schemaReporter = new SchemaReporter(
+      serverInfo,
+      executableSchema,
+      this.apiKey,
+      this.options.schemaReportingUrl,
+    );
+
+    const fallbackReportingDelayInMs = 20_000;
+
+    this.currentSchemaReporter = schemaReporter;
+    const logger = this.logger;
+
+    setTimeout(function() {
```

Comment:
It's a little awkward that `stop` doesn't clearTimeout this timeout or any of the ones in reportingLoop; even if they won't do anything, you can imagine a context where this will require a program (like a test?) to stay running for up to 10 seconds before doing nothing! Leaking setTimeout return values isn't a great habit to get into anyway. Why not make `inner` from `reportingLoop`  be a method on SchemaReporter, and save the return value from the active timeout on SchemaReporter, and clearTimeout on stop?